lib/angular_xss/safe_buffer.rb

```diff
@@ -1,20 +1,36 @@
 ActiveSupport::SafeBuffer.class_eval do
 
-  if private_method_defined? :html_escape_interpolated_argument
+  if private_method_defined?(:html_escape_interpolated_argument) || # Rails < 7
+    private_method_defined?(:explicit_html_escape_interpolated_argument) # Rails >= 7
 
     private
 
-    def html_escape_interpolated_argument_with_angular_xss(arg)
+    def explicit_html_escape_interpolated_argument_with_angular_xss(arg)
       if arg.html_safe?
         arg
       else
-        html_escape_interpolated_argument_without_angular_xss(AngularXss::Escaper.escape(arg))
+        explicit_html_escape_interpolated_argument_without_angular_xss(AngularXss::Escaper.escape(arg))
       end
```

Comment:
The implementation on at least Rails 6.1 and 7.0 is:

```ruby
        (!html_safe? || arg.html_safe?) ? arg : CGI.escapeHTML(arg.to_s)
```

We're missing the `!html_safe?` check on `self` here which might actually break supported features (I'm thinking of concatting `nil` onto a SafeBuffer which works in vanilla Rails; have not tested if that is broken here).

Might be worth checking if we can safely live without it, or if we need to implement a version switch, e.g. "Rails < 6 = only `arg.html_safe?` vs Rails 6+ = `!html_safe? || arg.html_safe?`).
If we have to do that, we could maybe do everything via version checks instead of checking for defined methods.

Reply:
The following snippet was always part of ActiveSupport and was just renamed:
```
 (!html_safe? || arg.html_safe?) ? arg : CGI.escapeHTML(arg.to_s)
# name in rails 7:explicit_html_escape_interpolated_argument
# name in rails 6.1 up to 4.1: html_escape_interpolated_argument
```
Even Rails 4.1 had the exact same implementation: https://github.com/rails/rails/blob/4-1-stable/activesupport/lib/active_support/core_ext/string/output_safety.rb#L233

I could not produce an issue with the current code (without `!html_safe?`):
```ruby
# vanilla
require 'active_support'
irb(main):009:0> str = ActiveSupport::SafeBuffer.new("")
=> ""
irb(main):010:0> str << nil
=> ""
irb(main):011:0> str << "<html>"
=> "&lt;html&gt;"

## with the monkey patch
require './lib/angular_xss/safe_buffer.rb'
require 'lib/angular_xss/escaper.rb'
str2 = ActiveSupport::SafeBuffer.new("")
=> ""
irb(main):015:0> str2 << nil
=> ""
str2 = ActiveSupport::SafeBuffer.new("")
=> ""
irb(main):015:0> str2 << nil
=> ""
 str << "{{foo}}"
=> "{{ $root.DOUBLE_LEFT_CURLY_BRACE }}foo}}"
```

..but I don't see why we should keep it this way either. I'll replace both usages with `(!html_safe? || arg.html_safe?)` and trust green specs on this one